resources/tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/resources/FilteredResourceTest.java

```diff
@@ -429,31 +431,35 @@ public void testCreateFilterOnLinkedFolderAndTarget2() throws Exception {
 		// // Refreshing restores sanity (hides the .cpp files)...
 		// folder.refreshLocal(IResource.DEPTH_INFINITE, getMonitor());
 		members = folder.members();
-		assertEquals("4.3", 1, members.length);
-		assertEquals("4.4", IResource.FILE, members[0].getType());
-		assertEquals("4.5", "foo.h", members[0].getName());
+		assertThat(members).hasSize(1).allSatisfy(member -> {
```

Comment:
for the specific case of asserting 1 element and then checking it's properties assertThat(collection).singleElement() can sometimes be helpful to shorten several lines to one. E.g. you can get from list to element to property via assertThat(collection).singleElement().extracting(fieldOfTheElement).isEqualTo(foo). Not sure if that is of any help here when you have multiple properties to be extracted. :)